persistence/src/commonMain/kotlin/com/wire/kalium/persistence/backup/ObfuscatedCopyExporter.kt

```diff
@@ -0,0 +1,180 @@
+ */
+package com.wire.kalium.persistence.backup
+
+import com.wire.kalium.persistence.dao.UserIDEntity
+import com.wire.kalium.persistence.db.PlatformDatabaseData
+import com.wire.kalium.persistence.db.UserDBSecret
+import com.wire.kalium.persistence.db.UserDatabaseBuilder
+import com.wire.kalium.persistence.db.createEmptyDatabaseFile
+import com.wire.kalium.persistence.db.nuke
+import com.wire.kalium.persistence.db.userDatabaseDriverByPath
+import com.wire.kalium.persistence.kaliumLogger
+import com.wire.kalium.util.DelicateKaliumApi
+import com.wire.kalium.util.FileUtil
+import com.wire.kalium.util.KaliumDispatcher
+import com.wire.kalium.util.KaliumDispatcherImpl
+import kotlinx.coroutines.withContext
+
+@DelicateKaliumApi("This class is used for debugging purposes only")
+class ObfuscatedCopyExporter internal constructor(
+    user: UserIDEntity,
+    private val platformDatabaseData: PlatformDatabaseData,
+    private val localDatabase: UserDatabaseBuilder,
+    private val kaliumDispatcher: KaliumDispatcher = KaliumDispatcherImpl
+) {
+    private val emptyPlainFIleNameUserId = user.copy(value = ("obfuscated-backup-" + user.value))
+
+    fun deleteCopyFile() {
+        nuke(emptyPlainFIleNameUserId, platformDatabaseData)
+    }
+
+    @Suppress("TooGenericExceptionCaught", "ReturnCount")
+    suspend fun exportToPlainDB(): String? {
+
+        val emptyPlainFilePath: String = createEmptyDatabaseFile(platformDatabaseData, emptyPlainFIleNameUserId) ?: return null
+
+        attachAndExport(localDatabase, emptyPlainFilePath)
+        val obfuscateResult = obfuscatePlainCopy(emptyPlainFilePath)
+
+        return if (obfuscateResult) {
+            emptyPlainFilePath
+        } else {
+            FileUtil.deleteDirectory(emptyPlainFilePath)
+            null
+        }
+    }
+
+    private suspend fun attachAndExport(localDatabase: UserDatabaseBuilder, emptyPlainFilePath: String) = withContext(kaliumDispatcher.io) {
+        localDatabase.sqlDriver.execute(
+            null,
+            "ATTACH DATABASE ? AS $OBFUSCATED_PLAIN_DB KEY ''",
+            0
+        ) {
+            bindString(0, emptyPlainFilePath)
+        }
+
+        localDatabase.sqlDriver.executeQuery(
+            null,
+            "SELECT sqlcipher_export('$OBFUSCATED_PLAIN_DB')",
+            { cursor -> cursor.next() },
+            0
+        )
+
+        localDatabase.sqlDriver.execute(
+            null,
+            "DETACH DATABASE $OBFUSCATED_PLAIN_DB",
+            0
+        )
+    }
+
+    @Suppress("TooGenericExceptionCaught", "LongMethod")
+    private suspend fun obfuscatePlainCopy(
+        plainDBPath: String,
+    ): Boolean = withContext(kaliumDispatcher.io) {
+        val plainDBDriver = userDatabaseDriverByPath(
+            platformDatabaseData,
+            plainDBPath,
+            UserDBSecret(ByteArray(0)),
+            false
+        )
+
+        try {
+            plainDBDriver.execute(
```

Comment:
those needed to be this way since opening this db file with sqlDelight resulted to errors where sqlDelight will try to create tables that have been already created via the export function